contracts/bonding/BondingManager.sol

```diff
@@ -127,20 +116,10 @@ contract BondingManager is IBondingManager, Manager {
         _;
     }
 
-    /*
-     * @dev BondingManager constructor. Sets a pre-existing address for the LivepeerToken contract
-     * @param _token LivepeerToken contract address
-     */
-    function BondingManager(
-        address _registry,
-        address _token,
-        uint256 _numActiveTranscoders,
-        uint64 _unbondingPeriod
-    )
-        Manager(_registry)
-    {
-        // Set LivepeerToken address
-        token = LivepeerToken(_token);
+    function BondingManager(address _controller) Manager(_controller) {}
+
+    function initialize(uint64 _unbondingPeriod, uint256 _numActiveTranscoders) external beforeInitialization returns (bool) {
```

Comment:
When we instantiate a target contract using its constructor we update the storage of the target contract. However, we want to update the storage of the proxy contract. So, the logic of the constructor is now moved into this `initialize` function that the proxy contract can call and execute it its own context to modify the storage of the proxy contract.